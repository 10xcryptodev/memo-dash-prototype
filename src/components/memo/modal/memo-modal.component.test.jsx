import React from 'react'
import { shallow } from 'enzyme'
import { Modal } from 'semantic-ui-react'

import MemoModalComponent from './memo-modal.component'
import testUsers from '../../../test-utils/test-users'

describe('<MemoModalComponent />', () => {
  let wrapper
  const alice = testUsers['alice']

  beforeEach(() => {
    const div = document.createElement('div')
    document.body.appendChild(div)
  })

  describe('should render', () => {
    it('without crashing', () => {
      wrapper = shallow(<MemoModalComponent />)
      expect(wrapper).toMatchSnapshot()
    })

    it('not render memo if not opened', () => {
      wrapper = shallow(<MemoModalComponent memo={alice.memos[0]} opened={false} />)
      expect(wrapper).toMatchSnapshot()
    })

    it('render memo if opened', () => {
      wrapper = shallow(<MemoModalComponent memo={alice.memos[0]} opened={true} />)
      expect(wrapper).toMatchSnapshot()
    })
  })

  describe('interaction', () => {
    it('should call onModalCloseClicked when clicked on close', () => {
      const onModalCloseClicked = jest.fn()
      wrapper = shallow(
        <MemoModalComponent onModalCloseClicked={onModalCloseClicked} memo={alice.memos[0]} opened={true} />
      )
      wrapper.find(Modal).simulate('close')
      expect(onModalCloseClicked).toHaveBeenCalled()
    })
  })
})
