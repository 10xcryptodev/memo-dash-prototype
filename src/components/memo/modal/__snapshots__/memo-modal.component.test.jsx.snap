// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MemoModalComponent /> should render not render memo if not opened 1`] = `""`;

exports[`<MemoModalComponent /> should render render memo if opened 1`] = `
<Segment
  className="memo"
>
  <CommentGroup
    threaded={true}
  >
    <Modal
      centered={false}
      className="memo-modal"
      closeIcon={true}
      closeOnDimmerClick={true}
      closeOnDocumentClick={false}
      dimmer={true}
      eventPool="Modal"
      open={true}
    >
      <ModalHeader>
        Memo
      </ModalHeader>
      <ModalContent>
        <CommentGroup>
          <Connect(MemoComponent)
            memo={
              Object {
                "avatarUrl": "https://s3.amazonaws.com/uifaces/faces/twitter/madcampos/128.jpg",
                "idx": 1,
                "memoDatetime": "2018-07-26T16:44:56.000Z",
                "memoLikesCount": 1,
                "memoRepliesCount": 0,
                "memoText": "Suscipit vel nobis.",
                "memoTipTotal": 0,
                "username": "alice",
              }
            }
            showReplies={true}
          />
        </CommentGroup>
      </ModalContent>
    </Modal>
  </CommentGroup>
</Segment>
`;

exports[`<MemoModalComponent /> should render without crashing 1`] = `""`;
